src/main/scala/reporting/DomainInterpreter.scala

```diff
@@ -0,0 +1,59 @@
+package viper.silicon.reporting
+import viper.silicon.reporting.{Converter,ExtractedModelEntry}
+import viper.silicon.state.terms._
+import viper.silicon.state.Identifier
+/**
+  * the most Generic form of interpreter
+  *  @param	E : input type
+  *  @param I : info object type
+  *  @param	T : result type
+  */
+trait AbstractInterpreter[E,I,T]{ 
+	def interpret(entry:E,info:I):T
+}
+trait ModelInterpreter[T,I] extends AbstractInterpreter[ExtractedModelEntry,I,T]
+trait DomainInterpreter[T,I] extends AbstractInterpreter[DomainValueEntry,I,T]
+
+
+case class IdentityInterpreter() extends ModelInterpreter[ExtractedModelEntry,Any]{
```

Comment:
Spacing:
* `...] {`
* `entry: ExtractedModelEntry`
* `... ExtractedModelEntry, anything ...`
* `... ExtractedModelEntry = entry`

Submitting code that's formatted similarly to the already existing code makes the code base more easily readable, when changing contributors are involved.

Please check all code for similar changes.